server/src/immich/controllers/library.controller.ts

```diff
@@ -23,11 +24,18 @@ export class LibraryController {
   constructor(private service: LibraryService) {}
 
   @Get()
-  getLibraries(@Auth() auth: AuthDto): Promise<ResponseDto[]> {
+  getUserLibraries(@Auth() auth: AuthDto): Promise<ResponseDto[]> {
     return this.service.getAllForUser(auth);
   }
 
+  @Post('/all')
+  @AdminRoute()
+  getAllLibraries(@Auth() auth: AuthDto, @Body() dto: SearchLibraryDto): Promise<ResponseDto[]> {
+    return this.service.getAll(auth, dto);
+  }
+
   @Post()
+  @AdminRoute()
```

Comment:
Why do you have admin checks inside of this route if it is already unreachable to non-admins because of this guard?

Reply:
Done, we're only relying on the controller guard now